daemon/api_systems_test.go

```diff
@@ -574,3 +574,98 @@ func (s *apiSuite) TestSystemActionNonRoot(c *check.C) {
 		"type":        "error",
 	})
 }
+
+func (s *apiSuite) TestSystemRebootNeedsRoot(c *check.C) {
+	restore := MockDeviceManagerReboot(func(dm *devicestate.DeviceManager, systemLabel, mode string) error {
+		c.Fatalf("request reboot should not get called")
+		return nil
+	})
+	defer restore()
+
+	body := `{"action":"reboot"}`
+	url := "/v2/systems"
+	req, err := http.NewRequest("POST", url, strings.NewReader(body))
+	c.Assert(err, check.IsNil)
+	req.RemoteAddr = "pid=100;uid=1000;socket=;"
+
+	rec := httptest.NewRecorder()
+	systemsActionCmd.ServeHTTP(rec, req)
+	c.Check(rec.Code, check.Equals, 401)
+}
+
+func (s *apiSuite) TestSystemRebootHappy(c *check.C) {
+	s.daemon(c)
+
+	for _, tc := range []struct {
+		systemLabel, mode string
+	}{
+		{"", ""},
+		{"20200101", ""},
+		{"", "recover"},
```

Comment:
what about additional cases of 

```suggestion
		{"", "recover"},
		{"", "run"},
		{"20200101", "run"},
```

Reply:
added